### DISCUSSION TITLE
Tarn.js pool settings?

### DISCUSSION THREAD

**Discussion by josias-r** [directus/directus#7981] [Archive]
I have recently updated some projects to the latest directus version. I've noticed that there is now a `DB_POOL_*` env variable.
Since I'm no DB expert, I'd like to clarify what this pooling option does.

My projects use PostgreSQL hosted on DigitalOcean. I used to have problems with connection limits reached, to which the solution was simply to use a connection pool on DigitalOcean's side. Is this now somehow replaced by Directus itself? Is using PostgreSQL connection Pools still necessary/recommended?

It seems strange to me that the client (directus) has pooling options. I always thought this needs to be on the DB side.

Some info and clarifications would be awesome :D

**Comment by rijkvanzanten** [comment]
> I'd like to clarify what this pooling option does.

Pooling in the context of Directus means how many database connections Directus itself will create/maintain to the database. The more connections are available to Directus, the faster parallel operations can be executed.

> I used to have problems with connection limits reached, to which the solution was simply to use a connection pool on DigitalOcean's side

The connection pool on DigitalOcean allows you to effectively reduce the number of used "connections" by load balancing the connections from Directus to the database to a single connection between pgBouncer and Postgres, so for example:

Directus can have 10 connections to pgBouncer
pgBouncer load balances those 10 over 2 actual connections to the database.

> It seems strange to me that the client (directus) has pooling options. I always thought this needs to be on the DB side.

At the end of the day, they are different things 🙂  Directus' pooling options allow you to set limits to how many connections Directus tries to open to the database at any given time, while the databases' pooling allows you to "virtually" increase the amount of connections your database can handle at a time.

**Reply by josias-r** [reply]
Awesome thanks for explaining!

Are the connection pool settings for Directus applying for every connected client to the instance? (Meaning more users means more connections)
Or is the rule running on the Directus instance itself? (Meaning more users will still use only a limited amount of connections)

**Reply by rijkvanzanten** [reply]
> Are the connection pool settings for Directus applying for every connected client to the instance? (Meaning more users means more connections) Or is the rule running on the Directus instance itself?

A "connected client" in that case is a single running instance of Directus itself, so you're right: more users will still use only a limited amount of connections
